### DISCUSSION TITLE
Owner M

### DISCUSSION THREAD

**Discussion by Mannyflow7** [reown-com/appkit#2992] [General]
const modal = createAppKit({
  adapters: [wagmiAdapter],
  projectId,
  networks: [mainnet, arbitrum],
  metadata: metadata,
  features: {
    swaps: false // Optional - true by default
  }
})import { createAppKit } from '@reown/appkit/react'
import { SolanaAdapter } from '@reown/appkit-adapter-solana/react'
import { solana, solanaTestnet, solanaDevnet } from '@reown/appkit/networks'
import { PhantomWalletAdapter, SolflareWalletAdapter } from '@solana/wallet-adapter-wallets'

// 0. Set up Solana Adapter
const solanaWeb3JsAdapter = new SolanaAdapter({
  wallets: [new PhantomWalletAdapter(), new SolflareWalletAdapter()]
})

// 1. Get projectId from https://cloud.reown.com
const projectId = 'YOUR_PROJECT_ID'

// 2. Create a metadata object - optional
const metadata = {
  name: 'AppKit',
  description: 'AppKit Solana Example',
  url: 'https://example.com', // origin must match your domain & subdomain
  icons: ['https://avatars.githubusercontent.com/u/179229932']
}

// 3. Create modal
createAppKit({
  adapters: [solanaWeb3JsAdapter],
  networks: [solana, solanaTestnet, solanaDevnet],
  metadata: metadata,
  projectId,
  features: {
    analytics: true // Optional - defaults to your Cloud configurationhttps://cloud.reown.comhttps://example.comhttps://avatars.githubusercontent.com/u/179229932wc:0b8294b7e9c8cc0967ecc8d4bfbb8aed6cf62c29e55a52558fa938d0fbac290d@2?expiryTimestamp=1727944504&methods=wc_sessionAuthenticate&relay-protocol=irn&symKey=a61b50ec8668f79acd092689816abd1d02282fa6125da54b6cc88afcc4d329ad0x61a828cc33d4e6e1b978218d6ddbe59b9db6399b

**Comment by Mannyflow7** [comment]
> const modal = createAppKit({
>   adapters: [wagmiAdapter],
>   projectId,
>   networks: [mainnet, arbitrum],
>   metadata: metadata,
>   features: {
>     swaps: false // Optional - true by default
>   }
> })import { createAppKit } from '@reown/appkit/react'
> import { SolanaAdapter } from '@reown/appkit-adapter-solana/react'
> import { solana, solanaTestnet, solanaDevnet } from '@reown/appkit/networks'
> import { PhantomWalletAdapter, SolflareWalletAdapter } from '@solana/wallet-adapter-wallets'
> 
> // 0. Set up Solana Adapter
> const solanaWeb3JsAdapter = new SolanaAdapter({
>   wallets: [new PhantomWalletAdapter(), new SolflareWalletAdapter()]
> })
> 
> // 1. Get projectId from https://cloud.reown.com
> const projectId = 'YOUR_PROJECT_ID'
> 
> // 2. Create a metadata object - optional
> const metadata = {
>   name: 'AppKit',
>   description: 'AppKit Solana Example',
>   url: 'https://example.com', // origin must match your domain & subdomain
>   icons: ['https://avatars.githubusercontent.com/u/179229932']
> }
> 
> // 3. Create modal
> createAppKit({
>   adapters: [solanaWeb3JsAdapter],
>   networks: [solana, solanaTestnet, solanaDevnet],
>   metadata: metadata,
>   projectId,
>   features: {
>     analytics: true // Optional - defaults to your Cloud configurationhttps://cloud.reown.comhttps://example.comhttps://avatars.githubusercontent.com/u/179229932wc:0b8294b7e9c8cc0967ecc8d4bfbb8aed6cf62c29e55a52558fa938d0fbac290d@2?expiryTimestamp=1727944504&methods=wc_sessionAuthenticate&relay-protocol=irn&symKey=a61b50ec8668f79acd092689816abd1d02282fa6125da54b6cc88afcc4d329ad0x61a828cc33d4e6e1b978218d6ddbe59b9db6399b

**Reply by Mannyflow7** [reply]
import { createAppKit } from '@reown/appkit/react'
import { SolanaAdapter } from '@reown/appkit-adapter-solana/react'
import { solana, solanaTestnet, solanaDevnet } from '@reown/appkit/networks'
import { PhantomWalletAdapter, SolflareWalletAdapter } from '@solana/wallet-adapter-wallets'

// 0. Set up Solana Adapter
const solanaWeb3JsAdapter = new SolanaAdapter({
  wallets: [new PhantomWalletAdapter(), new SolflareWalletAdapter()]
})

// 1. Get projectId from https://cloud.reown.com
const projectId = 'YOUR_PROJECT_ID'

// 2. Create a metadata object - optional
const metadata = {
  name: 'AppKit',
  description: 'AppKit Solana Example',
  url: 'https://example.com', // origin must match your domain & subdomain
  icons: ['https://avatars.githubusercontent.com/u/179229932']
}

// 3. Create modal
createAppKit({
  adapters: [solanaWeb3JsAdapter],
  networks: [solana, solanaTestnet, solanaDevnet],
  metadata: metadata,
  projectId,
  features: {
    analytics: true // Optional - defaults to your Cloud configurationconst modal = createAppKit({
  adapters: [wagmiAdapter],
  projectId,
  networks: [mainnet, arbitrum],
  metadata: metadata,
  features: {
    swaps: false // Optional - true by default
  }
})https://avatars.githubusercontent.com/u/179229932wc:0b8294b7e9c8cc0967ecc8d4bfbb8aed6cf62c29e55a52558fa938d0fbac290d@2?expiryTimestamp=1727944504&methods=wc_sessionAuthenticate&relay-protocol=irn&symKey=a61b50ec8668f79acd092689816abd1d02282fa6125da54b6cc88afcc4d329ad0x61a828cc33d4e6e1b978218d6ddbe59b9db6399b17279445047

**Comment by rtomas** [comment]
do you have questions ?
